request-manager/graph/grapher.go

```diff
@@ -244,26 +265,27 @@ func buildSeqGraph(seqSpec *spec.Sequence, idgen id.Generator) (seqGraph *Graph,
 					return nil, nil, fmt.Errorf("node %s missing job args: %s", nodeSpec.Name, strings.Join(missingArgs, ", "))
 				}
 
-				// Insert component into graph
+				// Insert node into graph
 				if len(nodeSpec.Dependencies) == 0 {
 					// Case: no dependencies; insert directly after start node
-					err := seqGraph.InsertComponentBetween(component, seqGraph.Source, seqGraph.Sink)
+					err := seqGraph.InsertComponentBetween(node, seqGraph.Source, seqGraph.Sink)
 					if err != nil {
 						return nil, nil, err
 					}
 				} else {
-					// Case: dependencies exist; insert between all its dependencies and the end node
+					// Case: dependencies exist; insert between all its
+					// dependencies and the end node
 					for _, dependencyName := range nodeSpec.Dependencies {
 						prevComponent := nodes[dependencyName]
-						err := seqGraph.InsertComponentBetween(component, prevComponent.Source, seqGraph.Sink)
+						err := seqGraph.InsertComponentBetween(node, prevComponent.Sink, seqGraph.Sink)
 						if err != nil {
 							return nil, nil, err
 						}
 					}
 				}
 
-				// Add to topological ordering
-				seqGraph.Order = append(seqGraph.Order, component.Source)
+				// Add this node to topological ordering
+				seqGraph.Order = append(seqGraph.Order, node.Source)
```

Comment:
Same comment as above: in next iteration of Graph, I hope to see this kind of low-level graph handling logic hidden at this level of code (i.e. caller of graph shouldn't have to do or know anything about graph's topo order).